main/ertp/api/contract-hosts.md

```diff
@@ -0,0 +1,158 @@
+# Contract Hosts
+
+ContractHost provides a way to install and run verifiable contracts.
+
+Install(source) creates Installations. Installations represent particular
+contract forms, which can be spawn()'d to create Contract instances
+initialized with specific terms. These Contracts manage invitations for
+multiple seats, each of which repesents a "role" in an interaction.
+
+Some seats provide the methods `getWinnings()` and `getRefunds()`, which return
+promises for payments for the outputs of the interaction. We provide the
+`collect()` method (described later) to simplify collection of winnings and
+refunds into appropriate purses.
+
+## contractHost.install(contractSrc)
+- `contractSrc` `{String}` - contractSrc is the source code for a Contract. See Contract for details.
+- Returns: `{Installation}`
+
+Install the source code for a Contract.
+
+```js
+const someEscrowInstallation = E(someHost).install(someEscrowExchangeSrc);
+```
+
+## contractHost.getInstallationSourceCode(installation)
+- `installation` `{Installation}`
+- Returns: `{String}`
+
+Get the actual source code of the Installation.
+
+```js
+const theSourceCode = E(someHost).getInstallationSourceCode(someInstallation);
+```
+
+## contractHost.redeem(invite)
+- `invite` `{Payment}`
+- Returns: `{Object}`
+
+Seat invitations are packaged as payments from the Invite Assay. Redeeming an invite returns a seat object with an arbitrary interface (the interface is at the discretion of the contract) which supports interaction according to the terms.
+
+```js
+const someSeat = E(someHost).redeem(someInvite);
+```
+
+## contractHost.getInviteAssay()
+- Returns: `{Assay}`
+
+The assay allows holders of seat invitations to get exclusive access to a Seat.
+
+```js
+const assayWithExclusiveAccess = E(someHost).getInviteAssay();
+```
+
+## installation.spawn(terms)
+- `terms` `{Terms}`
+- Returns: `{Invites <Object>}`
+
+Create a new `InviteMaker`, then call the Contract's `start()` method and return its results. The results are often a collection of seat invitations for the seats in the contract, but see [`coveredCall`](https://github.com/Agoric/ERTP/blob/master/core/coveredCall.js) for an exception.
+
+```js
+import harden from '@agoric/harden';
+
+const terms = harden({ left: whatPartyAWants, right: whatPartyBWants });
+const invitePayments = E(someInstallation).spawn(terms);
+```
+
+## installation.checkUnits(installation, inviteUnits, terms)
+- `installation` `{Installation}`
+- `inviteUnits` `{Units}`
+- `terms` `{Terms}`
+- Returns:
+
+### <span style="color:red">What does this method return? It's not stated in the chainmail file and it's not very clear in the code usage.</span>
```

Comment:
@katelynsills dominique used this to help identify questions with some of these. I'm not very sure myself as I get most of the information from the chainmail file and source code. Can you clarify all the questions identified with the `<span style="color:red">` font? It's easy to see if you preview the site. thanks!

Reply:
Sure, thanks for marking it up this way. I can try to clarify, but @Chris-Hibbert has been the main person working with contract host. Zoe replaces some of the features of the contract host, so it might be a lower priority to get this part specifically right.

Reply:
checkUnits either returns undefined, or throws. It is called to verify that the particular contract being invoked matches the terms the user was expecting. If there's no match, then we must not proceed.

Reply:
Ok, that makes more sense. Thanks!